### PR TITLE
Adding scsi remove and add test for Avocado [v3]

### PR DESCRIPTION
**Pull request by narasimhan-v**
Signed-off-by: Narasimhan V sim@linux.vnet.ibm.com

### CODE REVIEW

**Comment by narasimhan-v** [comment]
Attaching `job.log`
[job_log.txt](https://github.com/avocado-framework/avocado-misc-tests/files/365289/job_log.txt)

File: io/disk/scsi_remove_add.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        exit(1)
+END
+
+[[ -z $pci_device ]] && pci_device="0000:01:00.0" 
+
+echo "PCI Device: $pci_device"
+device_list_output=($(ls -l /dev/disk/by-path/ | grep "$pci_device" | awk '{print $NF}'))
+if [[ -z $device_list_output ]]; then
+    avocado_debug "No Devices in PCI ID $pci_device"
+    exit
```
Comment (apahim) [inline io/disk/scsi_remove_add.sh:40]: Is this a good ending? Avocado is showing `PASS` as test result when one enters this `if`.

File: io/disk/scsi_remove_add.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    sleep 5
+    lsscsi
+    echo "$scsi_num deleted"
+    echo "adding $scsi_num back"
+    echo "scsi add-single-device $scsi_num_seperated" > /proc/scsi/scsi
+    sleep 5
+    lsscsi
+    echo "$scsi_num added back"
+    echo
+done
```
Comment (apahim) [inline io/disk/scsi_remove_add.sh:64]: When will the test actually fail? O don't see an `exit 1` anywhere...

File: io/disk/scsi_remove_add.sh (hunk 1 above)
Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:40]: We do not have a way to skip tests in bash scripts.
Can you suggest a better way to handle this ?

File: io/disk/scsi_remove_add.sh (hunk 2 above)
Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:64]: Test will fail when any of the commands in line 54 or line 59 fails.

File: io/disk/scsi_remove_add.sh (hunk 1 above)
Comment (apahim) [inline io/disk/scsi_remove_add.sh:40]: `exit 1`

File: io/disk/scsi_remove_add.sh (hunk 2 above)
Comment (apahim) [inline io/disk/scsi_remove_add.sh:64]: Ok, sorry, I didn't notice the `-e` in the shebang.

File: io/disk/scsi_remove_add.sh (hunk 1 above)
Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:40]: So assumption is, instead of `SKIP` we can `FAIL` ?

Comment (apahim) [inline io/disk/scsi_remove_add.sh:40]: Yes, we don't have skip. But it's your call... I'm just wondering what would be a better fit here (`PASS` or `FAIL`). I'm ok with both.

Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:40]: I did not want it to `FAIL`. So, `PASS` is ok for me, till a way to skip is introduced for bash scripts.

Comment (apahim) [inline io/disk/scsi_remove_add.sh:40]: Ok, nice.

**Comment by apahim** [comment]
LGTM

File: io/disk/scsi_remove_add.sh (hunk 1 above)
Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:40]: I would prefer a `PASS` over a `FAIL`

Comment (narasimhan-v) [inline io/disk/scsi_remove_add.sh:40]: @apahim Thanks

**Pull request merged**